### PR TITLE
Feature/authentication delegation

### PR DESCRIPTION
**Pull request by acn-dgopa**
- Added authentication verification in home controller and redirect to a specific url
- Added authorize attributes to new endpoints
- Added authentication controller to expose refresh endpoint
- Added unit tests for authenticaiton controller, valid and invalid bearer token tests for the new endpoints

## Related Issue(s)
- #144 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green
The XSRF implementation will be addressed in a new pull request

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Configuration/GeneralSettings.cs [hunk 1]
```diff
@@ -24,5 +24,21 @@ public class GeneralSettings
...
+            get
+            {
+                return Environment.GetEnvironmentVariable("GeneralSettings__SBLBaseAdress") ?? SBLBaseAdress;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the host name.
+        /// </summary>
+        public string HostName { get; set; }
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Configuration/GeneralSettings.cs:42]: The HostName of what? Should be more specifically named and described

**Review by jonkjetiloye — COMMENTED** [review]
Added some quick comments

File: test/Altinn.AccessManagement.Tests/Controllers/HomeControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+using Xunit;
+using static Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext;
+
+namespace Altinn.AccessManagement.Tests.Controllers
+{
+    /// <summary>
+    /// Test class for <see cref="HomeController"></see>
+    /// </summary>
+    [Collection("HomeController Tests")]
+    public class HomeControllerTest : IClassFixture<CustomWebApplicationFactory<HomeController>>
```
Comment (TheTechArch) [inline test/Altinn.AccessManagement.Tests/Controllers/HomeControllerTest.cs:26]: Should we have a test with cookie also? I belive we have som in apps. 

Just to make sure both works?

File: src/Altinn.AccessManagement.Core/Configuration/GeneralSettings.cs (hunk 1 above)
Comment (acn-dgopa) [reply src/Altinn.AccessManagement.Core/Configuration/GeneralSettings.cs]: changed the name to AccessManagementApplicationHostName

File: src/Altinn.AccessManagement/Controllers/HomeController.cs [hunk 3]
```diff
@@ -1,48 +1,88 @@
...
+                {
+                    HttpOnly = false // Make this cookie readable by Javascript.
+                });
+            }
+            else
+            {
+                HttpContext.Response.Cookies.Append("XSRF-TOKEN", tokens.RequestToken, new CookieOptions
+                {
+                    Secure = true,
+                    HttpOnly = false // Make this cookie readable by Javascript.
```
Comment (acn-dgopa) [inline src/Altinn.AccessManagement/Controllers/HomeController.cs:61]: Sonar cloud is not happy with this implementation and I was wondering if it is a good idea to retun a xsrf token through an api endpoint than returning it as a cookie ? @TheTechArch @jonkjetiloye

File: src/Altinn.AccessManagement/Controllers/AuthenticationController.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        private readonly GeneralSettings _settings;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationController"/> class
+        /// </summary>
+        public AuthenticationController(
+            IAuthenticationClient authenticationClient, 
+            IOptions<GeneralSettings> settings, 
+            IOptions<PlatformSettings> platformSettings,
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement/Controllers/AuthenticationController.cs:28]: platformSettings option is unused. Some overlap of value between general and platform settings which should be cleaned up?
e.g. PlatformSettings.JwtCookieName and GeneralSettings.RuntimeCookieName seems to be the same

File: src/Altinn.AccessManagement/Controllers/HomeController.cs (hunk 3 above)
Comment (TheTechArch) [reply src/Altinn.AccessManagement/Controllers/HomeController.cs:61]: @acn-dgopa I am not sure that works. This is something related to how the react frontend (ajax framework) expects its to be.

**Review by TheTechArch — APPROVED** [review]

**Pull request merged**